frontend/viewer/src/lib/components/ui/format/format-duration.ts

```diff
@@ -0,0 +1,42 @@
+import {locale} from 'svelte-i18n-lingui';
+import {fromStore} from 'svelte/store';
+
+const currentLocale = fromStore(locale);
+type Duration = {
+  hours: number,
+  minutes: number,
+  seconds: number,
+  milliseconds: number,
+};
+export function formatDuration(value: Partial<Duration>, smallestUnit?: 'hours' | 'minutes' | 'seconds' | 'milliseconds', options?: {
+  style?: 'long' | 'short' | 'narrow' | 'digital',
+  hoursDisplay?: 'auto' | 'always',
+  minutesDisplay?: 'auto' | 'always',
+  secondsDisplay?: 'auto' | 'always',
+  millisecondsDisplay?: 'auto' | 'always'
+}) {
+
+  // @ts-expect-error TS2339
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment,@typescript-eslint/no-unsafe-call
+  const formatter = new Intl.DurationFormat(currentLocale.current, options);//has been polyfilled in main.ts
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-return,@typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access
+  return formatter.format(normalizeDuration(value, smallestUnit));
+}
+
+export function normalizeDuration(value: Partial<Duration>, smallestUnit?: 'hours' | 'minutes' | 'seconds' | 'milliseconds'): Partial<Duration>
+export function normalizeDuration(value: Partial<Duration>, smallestUnit?: 'seconds'): Omit<Duration, 'milliseconds'>
+export function normalizeDuration(value: Partial<Duration>): Duration
+export function normalizeDuration(value: Partial<Duration>, smallestUnit?: 'hours' | 'minutes' | 'seconds' | 'milliseconds'): Partial<Duration> {
+  let distanceMs = (value.hours ?? 0) * 3600000 + (value.minutes ?? 0) * 60000 + (value.seconds ?? 0) * 1000 + (value.milliseconds ?? 0);
```

Comment:
I'd like to put `_` characters in the large number literals to make it easier to see at a glance that they're correct. 1000 is easy enough, but I'd like `3600000` and `60000` to become `3_600_000` and `60_000` respectively.

Those are from ES2021 (https://caniuse.com/mdn-javascript_grammar_numeric_separators) so should be supported in all current browsers, and if our compilation step targets ES2020 or earlier I expect it to be fully capable of removing the `_` in compilation. But we'd want to double check the code output at least once just to make sure we're not preventing five-year-old Android phones from being able to load our site.